skimage/measure/simple_metrics.py

```diff
@@ -142,3 +144,47 @@ def compare_psnr(im_true, im_test, data_range=None):
 
     err = compare_mse(im_true, im_test)
     return 10 * np.log10((data_range ** 2) / err)
+
+
+def compare_nmi(im_true, im_test, *, bins=100):
+    """Compute the normalized mutual information.
+
+    The normalized mutual information is given by:
+
+                H(A) + H(B)
+      Y(A, B) = -----------
+                  H(A, B)
+
+    where H(X) is the entropy ``- sum(x log x) for x in X``.
+
+    It was proposed to be useful in registering images by Colin Studholme and
+    colleagues [1]_.
+
+    Parameters
+    ----------
+    im_true, im_test : ndarray
+        Images to be compared.
+    bins : int or sequence of int, optional
+        The number of bins along each axis of the joint histogram.
+
+    Returns
+    -------
+    nmi : float
+        The normalized mutual information between the two arrays, computed at
+        the granularity given by ``bins``.
+
+    References
+    ----------
+    .. [1] C. Studholme, D.L.G. Hill, & D.J. Hawkes (1999). An overlap
+           invariant entropy measure of 3D medical image alignment.
+           Pattern Recognition 32(1):71-86
+           :DOI:`10.1016/S0031-3203(98)00091-0`
+    """
+    hist, bin_edges = np.histogramdd([np.ravel(im_true), np.ravel(im_test)],
```

Comment:
not sure what will happen if the images are different sizes

Reply:
Probably nothing good 😅